pkg/declarativeconfig/notifier.go

```diff
@@ -0,0 +1,47 @@
+package declarativeconfig
+
+// KeyValuePair represents pair of key and value.
+type KeyValuePair struct {
+	Key   string `yaml:"key,omitempty"`
+	Value string `yaml:"value,omitempty"`
+}
+
+// GenericConfig is representation of storage.Notifier_Generic that supports transformation from YAML.
+type GenericConfig struct {
+	Endpoint            string         `yaml:"endpoint,omitempty"`
+	SkipTLSVerify       bool           `yaml:"skipTLSVerify,omitempty"`
+	CACert              string         `yaml:"caCert,omitempty"`
```

Comment:
This is probably going to be the cert in PEM format, similar to what we have for auth providers in SAML configuration / user-pki:
```yaml
saml:
  cert: |
    < cert pem>
```

Maybe it's worthwhile to call out this as `CACertPEM`? Unless you had a filepath in mind, either way it should be more descriptive either within a comment or a name what exactly we want here (path vs PEM).